From: "Barak A. Pearlmutter" <barak+git@pearlmutter.net>
Date: Mon, 28 Oct 2024 14:09:09 +0000
Subject: call callback

A function is supposed to use a provided callback; fixed.

See bugs.debian.org/777190
---
 lib/mhash.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/lib/mhash.c b/lib/mhash.c
index 645ff9f..6e10737 100644
--- a/lib/mhash.c
+++ b/lib/mhash.c
@@ -405,7 +405,9 @@ WIN32DLL_DEFINE
 
 	size = mhash_get_block_size( td->algorithm_given);
 	
-	digest = mutils_malloc(size);
+	digest =
+	    hash_malloc(mhash_get_block_size
+			(td->algorithm_given));
 
 	if (digest==NULL)
 	{
